release.nix

```diff
@@ -125,47 +141,66 @@ let
 
   jobs = {
     inherit dockerImageArtifact;
-    native =
-      let filteredBuilds = mapAttrsRecursiveCond (a: !(isList a)) (path: value:
-        if (any (p: take (length p) path == p) onlyBuildOnDefaultSystem) then filter (s: !(elem s nonDefaultBuildSystems)) value else value)
-        (packagePlatforms project);
-      in (mapTestOn (__trace (__toJSON filteredBuilds) filteredBuilds));
-    musl64 = mapTestOnCross musl64 (packagePlatformsCross (filterProject noMusl64Build));
     ifd-pins = mkPins {
       inherit (sources) iohk-nix "haskell.nix";
       inherit nixpkgs;
       inherit (pkgs.haskell-nix) hackageSrc stackageSrc;
     };
+  } // (optionalAttrs linuxBuild (with linuxRelease; {
+    linux =
+      let filteredBuilds = mapAttrsRecursiveCond (a: !(isList a)) (path: value:
+        if (any (p: take (length p) path == p) onlyBuildOnDefaultSystem) then filter (s: !(elem s nonDefaultBuildSystems)) value else value)
+        (packagePlatforms project);
+      in (mapTestOn (__trace (__toJSON filteredBuilds) filteredBuilds)) // {
+        # linux static builds:
+        musl64 = mapTestOnCross musl64 (packagePlatformsCross (filterProject noMusl64Build project));
+      };
+    cardano-node-linux = import ./nix/binary-release.nix {
+      inherit (linuxRelease) pkgs project;
+      platform = "linux";
+      exes = collectJobs jobs.linux.musl64.exes;
+    };
+  })) // (optionalAttrs macosBuild (with macosRelease; {
+    macos =
+      let filteredBuilds = mapAttrsRecursiveCond (a: !(isList a)) (path: value:
+        if (any (p: take (length p) path == p) onlyBuildOnDefaultSystem) then filter (s: !(elem s nonDefaultBuildSystems)) value else value)
+        (packagePlatforms project);
+      in (mapTestOn (__trace (__toJSON filteredBuilds) filteredBuilds));
     cardano-node-macos = import ./nix/binary-release.nix {
       inherit pkgs project;
       platform = "macos";
-      exes = filter (p: p.system == "x86_64-darwin") (collectJobs jobs.native.exes);
-    };
-    cardano-node-linux = import ./nix/binary-release.nix {
-      inherit pkgs project;
-      platform = "linux";
-      exes = filter (p: p.system == "x86_64-linux") (collectJobs jobs.musl64.exes);
+      exes = collectJobs jobs.macos.exes;
     };
-  } // (optionalAttrs windowsBuild {
-    "${mingwW64.config}" = mapTestOnCross mingwW64 (packagePlatformsCross (filterProject noCrossBuild));
+  })) // (optionalAttrs windowsBuild (with windowsRelease; {
+    "${mingwW64.config}" = mapTestOnCross mingwW64 (packagePlatformsCross (filterProject noCrossBuild project));
```

Comment:
Rather than calling this attrset `x86_64-w64-mingw32`, how about just `windows`?